ui/apps/platform/src/Containers/NetworkGraph/flows/AdvancedFlowsFilter.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {
+    Select,
+    SelectGroup,
+    SelectOption,
+    SelectPosition,
+    SelectVariant,
+} from '@patternfly/react-core';
+
+import useMultiSelect from 'hooks/useMultiSelect';
+
+export type AdvancedFlowsFilterType = {
+    flows: string[];
+    directionality: string[];
+    protocols: string[];
+    ports: string[];
+};
+
+export type AdvancedFlowsFilterProps = {
+    filters: AdvancedFlowsFilterType;
+    setFilters: React.Dispatch<React.SetStateAction<AdvancedFlowsFilterType>>;
+};
+
+export const defaultAdvancedFlowsFilters: AdvancedFlowsFilterType = {
+    flows: [],
+    directionality: [],
+    protocols: [],
+    ports: [],
+};
+
+// This function is used to convert our filters data structure to an array of strings
+// which the Select component expects
+function filtersToSelections(filters: AdvancedFlowsFilterType) {
+    return Object.values(filters).reduce((acc, curr) => {
+        return [...acc, ...Object.values(curr)];
+    }, []);
+}
+
+// This function is used to convert the selections array of strings, which the Select component
+// expects, into our filters data structure
+function selectionsToFilters(selections: string[]) {
```

Comment:
Might be cleaner to pull these two functions out into a helper file, especially if you think they will eventually be reused in other filters in the new network graph.

Factoring out would allow for quick yet comprehensive unit tests of all filter type combinations, too, and allow fewer e2e tests for same code coverage.